src/hooks/mediaHooks/useDeleteMultipleMediaHook.tsx

```diff
@@ -0,0 +1,84 @@
+import { AxiosError } from "axios"
+import {
+  useMutation,
+  UseMutationOptions,
+  UseMutationResult,
+  useQueryClient,
+} from "react-query"
+
+import { LIST_MEDIA_DIRECTORY_FILES_KEY } from "constants/queryKeys"
+
+import { apiService } from "services/ApiService"
+
+import { MediaService } from "services"
+import { MiddlewareError } from "types/error"
+import { MediaDirectoryParams } from "types/folders"
+import { SelectedMediaDto } from "types/media"
+
+const deleteMultipleMedia = async (
+  { siteName }: MediaDirectoryParams,
+  selectedMediaDto: SelectedMediaDto[]
+) => {
+  const mediaService = new MediaService({ apiClient: apiService })
+
+  await selectedMediaDto
+    .map((deleteMedia) => {
+      const pathTokens = deleteMedia.filePath.split("/")
+
+      const reqParams = {
+        siteName,
+        mediaDirectoryName: pathTokens.slice(0, -1).join("%2F"),
+        fileName: pathTokens.pop(),
+        sha: deleteMedia.sha,
+      }
+
+      return reqParams
+    })
+    .reduce(
+      (acc, curr) =>
+        acc
+          .then(() => {
+            const { sha, ...rest } = curr
+            mediaService.delete(rest, { sha }).catch((error) => {
+              // We want to continue even if some of the media files fail to delete
+              // because there is no turning back now if we fail (the earlier files
+              // would have been deleted)
+              return error
+            }) as Promise<void>
+          })
+          // This wait is necessary to avoid the repo lock
+          .then(() => new Promise((resolve) => setTimeout(resolve, 500))),
```

Comment:
quick qn why cant we just do sequentially? why have this 500 ms wait?

Reply:
^ non blocking if this is just done after the return. Else blocking since we might have an edge case here right if the request takes longer than 500?

Reply:
This is inside a `.then()` call, so it would happen after the delete request is sent and has complete. This is for the backend to catch up between requests.

Reply:
we could have used await here tho right?